src/main/rcsplit/rcsplit.c

```diff
@@ -0,0 +1,291 @@
+static serialPort_t *rcsplitSerialPort = NULL;
+static rcsplit_switch_state_t switchStates[RCSPLIT_CHECKBOX_ITEM_COUNT];
+static rcsplit_state_e cameraState = RCSPLIT_STATE_UNKNOWN;
+
+static unsigned char crc_high_first(unsigned char *ptr, unsigned char len)
+{
+    unsigned char i; 
+    unsigned char crc=0x00;
+    while(len--) {
+        crc ^= *ptr++;  
+        for (i=8; i>0; --i) { 
+            if (crc & 0x80)
+                crc = (crc << 1) ^ 0x31;
+            else
+                crc = (crc << 1);
+        }
+    }
+    return (crc); 
+}
+
+static void sendCtrlCommand(rcsplit_ctrl_cmd_argument_e argument)
+{
+    if (!rcsplitSerialPort)
+        return ;
+
+    uint8_t tmpBuffer[4];
+    unsigned char uart_buffer[5] = {0};
+    unsigned char crc = 0;
+
+    // first calc crc with the request, 
+    // note: the componnents that need to crc is [header]+[command]+[argument]+[tail], 
+    // no need calc  with crc field, event the crc field is zero
+    tmpBuffer[0] = RCSPLIT_PACKET_HEADER;
+    tmpBuffer[1] = RCSPLIT_PACKET_CMD_CTRL;
+    tmpBuffer[2] = argument;
+    tmpBuffer[3] = RCSPLIT_PACKET_TAIL;
+
+    crc = crc_high_first(tmpBuffer, 4);
+
+    // build up a full request [header]+[command]+[argument]+[crc]+[tail]
+    uart_buffer[0] = RCSPLIT_PACKET_HEADER;
+    uart_buffer[1] = RCSPLIT_PACKET_CMD_CTRL;
+    uart_buffer[2] = argument;
+    uart_buffer[3] = crc;
+    uart_buffer[4] = RCSPLIT_PACKET_TAIL;
+
+    // write to device
+    serialWriteBuf(rcsplitSerialPort, uart_buffer, 5);
+}
+
+static void rcsplitProcessMode() 
+{
+    // if the device not ready, do not handle any mode change event
+    if (RCSPLIT_STATE_IS_READY != cameraState) 
+        return ;
+
+    for (rcsplitBoxId_e i = 0; i < RCSPLIT_CHECKBOX_ITEM_COUNT; i++) {
+        // simulate wifi button press action
+        if (IS_RCSPLIT_MODE_ACTIVE(i)) {
+            // check last state of this mode, if it's true, then ignore it. 
+            // Here is a logic to make a toggle control for this mode
+            if (switchStates[i].isActivited) {
+                continue;
+            }
+
+            uint8_t argument = RCSPLIT_BOX_INVALID;
+            switch (i) {
+                case RCSPLIT_BOX_SIM_WIFI_BUTTON:
+                    argument = 0x1;
+                    break;
+                case RCSPLIT_BOX_SIM_POWER_BUTTON:
+                    argument = 0x2;
+                    break;
+                case RCSPLIT_BOX_SIM_CHANGE_MODE:
+                    argument = 0x3;
+                    break;
+                default:
+                    argument = RCSPLIT_BOX_INVALID;
+                    break;
+            }
+            
+            if (argument != RCSPLIT_BOX_INVALID) {
+                sendCtrlCommand(argument);
+                switchStates[i].isActivited = true;
+            } else {
+            }
+        } else {
+            switchStates[i].isActivited = false;
+        }
+    }
+}
+
+rcsplit_state_e unitTestRCsplitState()
```

Comment:
I'm afraid code size will stay the same. gcc is clever enough to remove unused code ...